packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -391,13 +410,23 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 		);
 	}
 
-	private getBlobHandle(id: string): IFluidHandle<ArrayBufferLike> {
+	private getBlobHandle(id: string): BlobHandle {
 		assert(
 			this.redirectTable.has(id) || this.pendingBlobs.has(id),
 			0x384 /* requesting handle for unknown blob */,
 		);
-		return new BlobHandle(`${BlobManager.basePath}/${id}`, this.routeContext, async () =>
-			this.getBlob(id),
+		const entry = this.pendingBlobs.get(id) as PendingBlob;
+		return new BlobHandle(
+			`${BlobManager.basePath}/${id}`,
+			this.routeContext,
+			async () => this.getBlob(id),
+			() => {
+				// in detached container case, there is no pending entry
+				if (entry) {
```

Comment:
can we flip this an only specify the callback if there is an entry, rather than handling it in the callback itself?